File: app/Domain/Messages/MessageRequacked.php
<?php

namespace App\Domain\Messages;

use App\Domain\Identity\UserId;
use App\Domain\IDomainEvent;

class MessageRequacked implements IDomainEvent
{
    /** @var MessageId */
    private $messageId;

    /** @var UserId */
    private $requackerId;

    public function __construct(MessageId $messageId, UserId $requackerId)
    {
        $this->messageId = $messageId;
        $this->requackerId = $requackerId;
    }

    /**
     * @return MessageId
     */
    public function getMessageId()
    {
        return $this->messageId;
    }

    /**
     * @return UserId
     */
    public function getRequackerId()
    {
        return $this->requackerId;
    }

    /**
     * @return string
     */
    public function getAggregateId()
    {
        return $this->getMessageId()->getId();
    }
}

File: Mixter.Domain/Core/Messages/Events/MessageRequacked.cs
using Mixter.Domain.Identity;

namespace Mixter.Domain.Core.Messages.Events
{
    [Event]
    public struct MessageRequacked : IDomainEvent
    {
        public MessageId Id { get; private set; }

        public UserId Requacker { get; private set; }

        public MessageRequacked(MessageId id, UserId requacker)
            : this()
        {
            Id = id;
            Requacker = requacker;
        }

        public object GetAggregateId()
        {
            return Id;
        }
    }
}
